### PR TITLE
Issue #1367: Fixed default value sometimes set incorrectly for radios.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1367

### CODE REVIEW

**Comment by quicksketch** [comment]
Looks like we have a PHP notice in the test results:

```
Exception Notice     form.inc          2936 theme_radio()                      
    Undefined index: #return_value
```

File: core/includes/form.inc [hunk 1]
```diff
@@ -2932,9 +2932,24 @@ function theme_radio($variables) {
   $element['#attributes']['type'] = 'radio';
   element_set_attributes($element, array('id', 'name', '#return_value' => 'value'));
 
-  if (isset($element['#return_value']) && $element['#value'] !== FALSE && $element['#value'] == $element['#return_value']) {
-    $element['#attributes']['checked'] = 'checked';
+  debug($element['#value']);
+  debug($element['#return_value']);
```
Comment (quicksketch) [inline core/includes/form.inc]: Heh, we can't have debug() statements in here!

**Comment by quicksketch** [comment]
We should also add tests for this new set of conditions, as mentioned in the issue https://github.com/backdrop/backdrop-issues/issues/1367#issuecomment-168354712.

File: core/includes/form.inc (hunk 1 above)
Comment (jenlampton) [reply core/includes/form.inc]: heh, fixed.

**Comment by jenlampton** [comment]
I added some new tests, but I'm not entirely sure they are testing what I think they are testing.

**Comment by drupol** [comment]
What's the status of this issue ?

**Comment by jenlampton** [comment]
@drupol it looks like it can't be reproduced, see https://github.com/backdrop/backdrop-issues/issues/1367.  Closing this PR for now.

**Comment by drupol** [comment]
Cool, thanks !

**Pull request closed without merging**